providers/src/airflow/providers/fab/www/extensions/init_views.py

```diff
@@ -78,6 +108,59 @@ def validate_schema(self, data, url):
         return super().validate_schema(data, url)
 
 
+base_paths: list[str] = []  # contains the list of base paths that have api endpoints
+
+
+def init_api_error_handlers(app: Flask) -> None:
+    """Add error handlers for 404 and 405 errors for existing API paths."""
+
+    @app.errorhandler(404)
+    def _handle_api_not_found(ex):
+        if any([request.path.startswith(p) for p in base_paths]):
+            # 404 errors are never handled on the blueprint level
+            # unless raised from a view func so actual 404 errors,
+            # i.e. "no route for it" defined, need to be handled
+            # here on the application level
+            return common_error_handler(ex)
+        else:
+            from airflow.providers.fab.www.views import not_found
+
+            return not_found(ex)
+
+    @app.errorhandler(405)
+    def _handle_method_not_allowed(ex):
+        if any([request.path.startswith(p) for p in base_paths]):
+            return common_error_handler(ex)
+        else:
+            from airflow.providers.fab.www.views import method_not_allowed
+
+            return method_not_allowed(ex)
+
+    app.register_error_handler(ProblemException, common_error_handler)
+
+
+def init_api_connexion(app: Flask) -> None:
```

Comment:
We are probably at a point we can rip out the old api, to be honest - I think aip-84 is far enough along.

If we want to keep it for now, that's fine, however we might want to leave a breadcrumb that this needs to go before AF3.

Reply:
I agree. Though, removing it make some tests fail. Some tests in FAB provider are using Rest API endpoints (e.g. `/api/v1/pools` in `providers/tests/fab/auth_manager/api_endpoints/test_auth.py`). So we need to update these tests as well. I prefer doing it in a separate PR because I think this one is big enough already.